src/main/kotlin/calculator/StringAddCalculator.kt

```diff
@@ -0,0 +1,66 @@
+package calculator
+
+class StringAddCalculator {
+
+    fun calculate(input: String?): Int {
+
+        if (input.isNullOrEmpty()) {
+            return ZERO
+        } else if (input.length == 1 && input[0].isDigit()) {
+            return input.toInt()
+        } else if (!containsDelimiter(input) && input.toInt() < ZERO) {
+            throw RuntimeException("음수를 입력할 수 없습니다")
+        }
+
+        val stringList = split(input)
+
+        return stringAdd(stringList)
+    }
+
+    fun split(input: String): List<String> {
+
+        var customDelimiter = getCustomDelimiter(input)
+        var customString: String? = null
+
+        customDelimiter?.let {
+            customDelimiter = DELIMITER.plus("|$it")
+            customString = input.split("\n")[1]
+        }
+
+        val splitList = customDelimiter?.let { customString?.split(it.toRegex()) }
+            ?:input.split(DELIMITER.toRegex())
+
+        checkNegativeNumber(splitList)
+        return splitList
+    }
+
+    fun getCustomDelimiter(input: String): String? {
+        val result = Regex("//(.)\n(.*)").find(input)
+        return result?.let {
+            return it.groupValues[1]
+        }
+    }
+
+    private fun checkNegativeNumber(splitList: List<String>) {
+        splitList.forEach {
+            if (it.toInt() < ZERO) {
+                throw RuntimeException("음수를 입력할 수 없습니다")
+            }
+        }
+    }
+
+    fun stringAdd(input: List<String>): Int {
```

Comment:
```suggestion
    fun add(input: List<String>): Int {
```
함수명은 대체로 동사구의 형태로 작성하는 것을 권장합니다.
함수명 또한 마찬가지로, 자료형과 관련된 문구를 사용하지 않는 것을 권장합니다.
함수가 의미하는 바를 알기 쉽도록 함수명을 개선해보면 어떨까요?